libfdt/libfdt.h

```diff
@@ -525,9 +525,10 @@ int fdt_path_offset_namelen(const void *fdt, const char *path, int namelen);
  * address).
  *
  * If the path is not absolute (i.e. does not begin with '/'), the
- * first component is treated as an alias.  That is, the property by
- * that name is looked up in the /aliases node, and the value of that
- * property used in place of that first component.
+ * first component is treated as an alias (or, if it begins with '&',
+ * the rest is treated as a symbol).  That is, the property by that
+ * name is looked up in the /aliases (or /__symbols__) node, and the
+ * value of that property used in place of that first component.
  *
```

Comment:
Something that is missing here is that the components in `path` don't need to specify the unit address ...

Reply:
Huh? Try looking at the commit (or the "improve documentation for fdt_path_offset()") with `-U10`. Then you'll see that that is right there in the context:
```
--- a/libfdt/libfdt.h
+++ b/libfdt/libfdt.h
@@ -517,24 +517,49 @@ int fdt_path_offset_namelen(const void *fdt, const char *path, int namelen);
  * @fdt: pointer to the device tree blob
  * @path: full path of the node to locate
  *
  * fdt_path_offset() finds a node of a given path in the device tree.
  * Each path component may omit the unit address portion, but the
  * results of this are undefined if any such path component is
  * ambiguous (that is if there are multiple nodes at the relevant
  * level matching the given component, differentiated only by unit
  * address).
  *
+ * If the path is not absolute (i.e. does not begin with '/'), the
+ * first component is treated as an alias.  That is, the property by
+ * that name is looked up in the /aliases node, and the value of that
+ * property used in place of that first component.
```

Unless I'm misunderstanding what you were trying to say, I don't think there's anything to fix.

Reply:
Ah indeed. It's documented and I missed that. While working on dtc for my change I stumbled about this "feature" that is inherited to e.g. `fdt_overlay_target_offset()` which is surprising at least. (If you're subscribed to `devicetree-compiler@vger.kernel.org`, see my mail "Bug or feature? /somenode overwrites /somenode@17 in fdt-overlay" with Message-Id `20230425145504.tswuitufzikuhmnz@pengutronix.de`, I didn't find it in a web archive.)

Reply:
>  (If you're subscribed to `devicetree-compiler@vger.kernel.org`, see my mail "Bug or feature? /somenode overwrites /somenode@17 in fdt-overlay" with Message-Id `20230425145504.tswuitufzikuhmnz@pengutronix.de`, I didn't find it in a web archive.)

I'm not subscribed, but it seems that that ML should be archived on lore.kernel.org, and if it isn't already (I can't find it), someone should ask helpdesk to do it (see https://korg.docs.kernel.org/lore.html).